LPD-38011 Refactor BlogsPostingImageResourceImpl

marcogalluzzi opened the pull request:
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-38011

This [suggestion](https://github.com/brianchandotcom/liferay-portal/pull/154293#issuecomment-2359233798) comes from Brian Chan.

I'm refactoring the class `BlogsPostingImageResourceImpl` to avoid duplicated logic related to the access of `_portletFileRepository` that already exists in `BlogsEntryLocalServiceImpl`. 

:warning:  NOTE: I don't know if we need to add new permissions for the GET and DELETE methods in `BlogsEntryServiceImpl`.

# How am I fixing it?

Now all the logic is stored in one place: `BlogsEntryLocalServiceImpl`.

# How can you verify that it works?

`cd modules/apps/headless/headless-delivery/headless-delivery-test`
`gw tI --tests BlogPostingImageResourceTest`

`cd modules/apps/blogs/blogs-test`
`gw tI --tests "*testAddAttachmentFileEntryWith*"`

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-3.liferay.com/job/test-portal-source-format/8112/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-source-format/builds/8112/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/5624963c55511dc19881843e67901a3eff0efabc">5624963c55511dc19881843e67901a3eff0efabc</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/marcogalluzzi/liferay-portal/tree/LPD-38011_Refactor-BlogsPostingImageResourceImpl">LPD-38011_Refactor-BlogsPostingImageResourceImpl</a><br/>Branch GIT ID: <a href="https://github.com/marcogalluzzi/liferay-portal/commit/c7f9d69c9ee9079e6c02e3b7d7df4e41dac75b74">c7f9d69c9ee9079e6c02e3b7d7df4e41dac75b74</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-3.liferay.com/job/test-portal-source-format/8112/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-source-format/builds/8112/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-3.liferay.com/job/test-portal-source-format/8112/">test-portal-source-format#8112</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-3.liferay.com/userContent/jobs/test-portal-source-format/builds/8112/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-content-management/liferay-portal/pull/5938">liferay-content-management#5938</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/31706719">[master] ci:test:sf - marcogalluzzi &gt; liferay-content-management - PR#5938 - 2024-10-04[00:34:43]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>31706719<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-3.liferay.com/job/publish-testray-report/4485/">publish-testray-report#4485</a><br/></div></div>

adolfopa commented on modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java:
```diff
@@ -747,6 +767,32 @@ public BlogsEntry fetchEntry(long groupId, String urlTitle) {
+	}
+
+	@Override
+	public FileEntry getAttachmentFileEntryByExternalReferenceCode(
+			long groupId, String externalReferenceCode)
+		throws PortalException {
```
The order of the arguments is the reverse; see other occurrences of `get.*ByExternalReferenceCode` is `*ServiceImpl.java` or the usage of `getPortletFileEntryByExternalReferenceCode` just below.

marcogalluzzi replied:
OK, I will check, thanks

marcogalluzzi replied:
Order changed

adolfopa commented on modules/apps/blogs/blogs-test/src/testIntegration/java/com/liferay/blogs/service/test/BlogsEntryLocalServiceTest.java:
```diff
@@ -648,13 +660,20 @@ public void testDeleteDiscussion() throws Exception {
+		}
+		catch (Exception exception) {
+			Assert.assertTrue(exception instanceof NoSuchEntryException);
+		}
```
Q: Why this instead of the `expected` attribute in the annotation?

marcogalluzzi replied:
How do we know that the exception doesn't come from the previous line
```java
_blogsEntryLocalService.deleteEntry(entry);
```

marcogalluzzi commented:
Hi @adolfopa I've made the requested changes. The most significative are:

* I've added permission checks in `BlogsEntryService`
* I've added the missing tests in `BlogsEntryServiceTest`

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-40.liferay.com/job/test-portal-source-format/6445/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/6445/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/9e6ad0e4d897e2d5317231c46841151f8d8a1b32">9e6ad0e4d897e2d5317231c46841151f8d8a1b32</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/marcogalluzzi/liferay-portal/tree/LPD-38011_Refactor-BlogsPostingImageResourceImpl">LPD-38011_Refactor-BlogsPostingImageResourceImpl</a><br/>Branch GIT ID: <a href="https://github.com/marcogalluzzi/liferay-portal/commit/443c9d08d95b69d217b6acab55252e0e44cedf0d">443c9d08d95b69d217b6acab55252e0e44cedf0d</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-40.liferay.com/job/test-portal-source-format/6445/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/6445/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:stable - 25 out of 26 jobs passed</h3><h3>:x: ci:test:relevant - 31 out of 36 jobs passed in 3 hours 15 minutes</h3><details><summary>Click here for more details.</summary><br/><p>This pull is eligible for <a href="https://grow.liferay.com/share/CI+liferay-continuous-integration+GitHub+Commands#General-Commands">reevaluation</a>. When this <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)/919">upstream build</a> has completed, using the following CI command will compare this pull request result against a more recent upstream result:<pre>ci:reevaluate:1407401_11388</pre></p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/9e6ad0e4d897e2d5317231c46841151f8d8a1b32">9e6ad0e4d897e2d5317231c46841151f8d8a1b32</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/9a8f329d547a998b08d2d630bf042adef665bc86">9a8f329d547a998b08d2d630bf042adef665bc86</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/#/project/35392/routines/590307/build/41451711">EE Development Acceptance (master) - 918 - 2024-10-15[00:38:53]</a></p><details><summary><strong>ci:test:stable - 25 out of 26 jobs PASSED</strong></summary><div><h4>1 Failed Jobs:</h4><ul><li>central-requirements<ul><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1066100/">central-requirements/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57-stable<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450693/">functional-smoke-tomcat90-mysql57-stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-stable<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450691/">functional-upgrade-tomcat90-mysql57-stable/0/0</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450692/">functional-upgrade-tomcat90-mysql57-stable/0/1</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428762/">functional-upgrade-tomcat90-mysql57-stable/0/2</a></li></ul></li><li>js-unit<ul><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1102660/">js-unit/0/0</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1102661/">js-unit/1/0</a></li></ul></li><li>modules-compile<ul><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/435044/">modules-compile/0/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/383178/">modules-compile/0/1</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428765/">modules-compile/0/2</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/436846/">modules-compile/0/3</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/370295/">modules-compile/0/4</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450694/">modules-compile/0/5</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374449/">modules-compile/0/6</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/370294/">modules-compile/0/7</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/479659/">modules-compile/0/8</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450695/">modules-compile/0/9</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428763/">modules-compile/0/10</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/370296/">modules-compile/0/11</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/476420/">modules-compile/0/12</a></li></ul></li><li>modules-integration-mysql57-stable<ul><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962063/">modules-integration-mysql57-stable/0/0</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962062/">modules-integration-mysql57-stable/0/1</a></li></ul></li><li>modules-unit-stable<ul><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428764/">modules-unit-stable/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-stable<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450690/">playwright-js-tomcat90-mysql57-stable/0/0</a></li></ul></li><li>service-builder<ul><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1066101/">service-builder/0/0</a></li></ul></li><li>unit-stable<ul><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374450/"><strike>unit-stable/0/0</strike></a></li></ul></li></ul></div><details><summary><strong>25 Successful Jobs:</strong></summary><ul/></details></details><details><summary><strong>ci:test:relevant - 31 out of 36 jobs PASSED</strong></summary><div><h4>5 Failed Jobs:</h4><ul><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest(master)/11388/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li><li>central-requirements<ul><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1066100/">central-requirements/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57-stable<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450693/">functional-smoke-tomcat90-mysql57-stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-stable<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450691/">functional-upgrade-tomcat90-mysql57-stable/0/0</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450692/">functional-upgrade-tomcat90-mysql57-stable/0/1</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428762/">functional-upgrade-tomcat90-mysql57-stable/0/2</a></li></ul></li><li>js-unit<ul><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1102660/">js-unit/0/0</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1102661/">js-unit/1/0</a></li></ul></li><li>modules-compile<ul><li><a href="https://test-1-10.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/435044/">modules-compile/0/0</a></li><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/383178/">modules-compile/0/1</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428765/">modules-compile/0/2</a></li><li><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/436846/">modules-compile/0/3</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/370295/">modules-compile/0/4</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450694/">modules-compile/0/5</a></li><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374449/">modules-compile/0/6</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/370294/">modules-compile/0/7</a></li><li><a href="https://test-1-8.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/479659/">modules-compile/0/8</a></li><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450695/">modules-compile/0/9</a></li><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428763/">modules-compile/0/10</a></li><li><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/370296/">modules-compile/0/11</a></li><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/476420/">modules-compile/0/12</a></li></ul></li><li>modules-integration-mysql57<ul><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962064/">modules-integration-mysql57/0/0</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1114574/"><strike>modules-integration-mysql57/0/1</strike></a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300/"><strike>modules-integration-mysql57/0/2</strike></a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734/"><strike>modules-integration-mysql57/0/3</strike></a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1024638/">modules-integration-mysql57/0/4</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/999782/">modules-integration-mysql57/0/5</a></li></ul></li><li>modules-integration-mysql57-stable<ul><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962063/">modules-integration-mysql57-stable/0/0</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/962062/">modules-integration-mysql57-stable/0/1</a></li></ul></li><li>modules-semantic-versioning<ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/999752/">modules-semantic-versioning/0/0</a></li></ul></li><li>modules-unit<ul><li><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/383177/">modules-unit/0/0</a></li></ul></li><li>modules-unit-stable<ul><li><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/428764/">modules-unit-stable/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57<ul><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374448/">playwright-js-tomcat90-mysql57/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-stable<ul><li><a href="https://test-1-9.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/450690/">playwright-js-tomcat90-mysql57-stable/0/0</a></li></ul></li><li>service-builder<ul><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1066101/">service-builder/0/0</a></li></ul></li><li>unit-stable<ul><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374450/"><strike>unit-stable/0/0</strike></a></li></ul></li></ul></div><details><summary><strong>31 Successful Jobs:</strong></summary><ul/></details></details><h5>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11388/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734//consoleText">modules-integration-mysql57/0/3</a><ol><li><details><summary><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734/testReport/com.liferay.headless.delivery.resource.v1_0.test/KnowledgeBaseArticleResourceTest">com.liferay.headless.delivery.resource.v1_0.test.KnowledgeBaseArticleResourceTest</a> - <a>Failed 0 of last 1</a></summary><div><ol><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734//testReport/com.liferay.headless.delivery.resource.v1_0.test/KnowledgeBaseArticleResourceTest/testGetKnowledgeBaseArticleKnowledgeBaseArticlesPageWithFilterStringContains">testGetKnowledgeBaseArticleKnowledgeBaseArticlesPageWithFilterStringContains</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734//testReport/com.liferay.headless.delivery.resource.v1_0.test/KnowledgeBaseArticleResourceTest/testGetKnowledgeBaseFolderKnowledgeBaseArticlesPageWithSortString">testGetKnowledgeBaseFolderKnowledgeBaseArticlesPageWithSortString</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734//testReport/com.liferay.headless.delivery.resource.v1_0.test/KnowledgeBaseArticleResourceTest/testPatchKnowledgeBaseArticle">testPatchKnowledgeBaseArticle</a></li></ol></div></details></li><li><details><summary><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734/testReport/com.liferay.headless.delivery.resource.v1_0.test/DocumentResourceTest">com.liferay.headless.delivery.resource.v1_0.test.DocumentResourceTest</a> - <a>Failed 0 of last 1</a></summary><div><ol><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734//testReport/com.liferay.headless.delivery.resource.v1_0.test/DocumentResourceTest/testGetAssetLibraryDocumentsPageWithPagination">testGetAssetLibraryDocumentsPageWithPagination</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734//testReport/com.liferay.headless.delivery.resource.v1_0.test/DocumentResourceTest/testGetDocumentFolderDocumentsPage">testGetDocumentFolderDocumentsPage</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1121734//testReport/com.liferay.headless.delivery.resource.v1_0.test/DocumentResourceTest/testGetSiteDocumentsPageWithSortDateTime">testGetSiteDocumentsPageWithSortDateTime</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374450//consoleText">unit-stable/0/0</a><ol><li><details><summary><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374450/testReport/junit.framework/TestSuite">junit.framework.TestSuite</a></summary><div><ol><li><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/374450//testReport/junit.framework/TestSuite/com_liferay_portal_service_ServiceContextAdviceTest">com.liferay.portal.service.ServiceContextAdviceTest</a></li></ol></div></details></li></ol></div></li></ol><hr/><details><summary><strong>Failures in common with <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at 9a8f329d547a998b08d2d630bf042adef665bc86:</strong></summary><ol><li><div><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1114574//consoleText">modules-integration-mysql57/0/1</a><ol><li><details><summary><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1114574/testReport/com.liferay.headless.batch.engine.resource.v1_0.test/ExportTaskResourceTest">com.liferay.headless.batch.engine.resource.v1_0.test.ExportTaskResourceTest</a> - <a>Failed 1 of last 1</a></summary><div><ol><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1114574//testReport/com.liferay.headless.batch.engine.resource.v1_0.test/ExportTaskResourceTest/testPostExportTask">testPostExportTask</a></li></ol></div></details></li><li><details><summary><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1114574/testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest">com.liferay.portal.log.assertor.PortalLogAssertorTest</a></summary><div><ol><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1114574//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">testScanXMLLog</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300//consoleText">modules-integration-mysql57/0/2</a><ol><li><details><summary><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300/testReport/com.liferay.headless.builder.model.listener.test/APIPropertyRelevantObjectEntryModelListenerTest">com.liferay.headless.builder.model.listener.test.APIPropertyRelevantObjectEntryModelListenerTest</a> - <a>Failed 1 of last 1</a></summary><div><ol><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300//testReport/com.liferay.headless.builder.model.listener.test/APIPropertyRelevantObjectEntryModelListenerTest/testAddAPIProperty">testAddAPIProperty</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300//testReport/com.liferay.headless.builder.model.listener.test/APIPropertyRelevantObjectEntryModelListenerTest/testAddRecordAPIProperty">testAddRecordAPIProperty</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300//testReport/com.liferay.headless.builder.model.listener.test/APIPropertyRelevantObjectEntryModelListenerTest/testRemoveAPIProperty">testRemoveAPIProperty</a></li></ol></div></details></li><li><details><summary><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300/testReport/com.liferay.headless.delivery.resource.v1_0.test/ContentTemplateResourceTest">com.liferay.headless.delivery.resource.v1_0.test.ContentTemplateResourceTest</a> - <a>Failed 0 of last 1</a></summary><div><ol><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300//testReport/com.liferay.headless.delivery.resource.v1_0.test/ContentTemplateResourceTest/testGetSiteContentTemplatesPageWithFilterStringEquals">testGetSiteContentTemplatesPageWithFilterStringEquals</a></li></ol></div></details></li><li><details><summary><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300/testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest">com.liferay.portal.log.assertor.PortalLogAssertorTest</a></summary><div><ol><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/960300//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">testScanXMLLog</a></li></ol></div></details></li></ol></div></li></ol></details></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest(master)/11388/">test-portal-acceptance-pullrequest(master)#11388</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11388/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-content-management/liferay-portal/pull/5938">liferay-content-management#5938</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/41660284">[master] ci:test:relevant - marcogalluzzi &gt; liferay-content-management - PR#5938 - 2024-10-15[03:57:47]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>41660284<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-40.liferay.com/job/publish-testray-report/24639/">publish-testray-report#24639</a><br/></div></div>

marcogalluzzi commented:
Failing tests are not related. Forcing the forwarding.

marcogalluzzi commented:
ci:forward:force

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` if the following test suites complete:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
AND If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
Skipping previously completed test suites:
- `ci:test:relevant`
- `ci:test:sf`

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>stable</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest(master)/10276/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-30.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/10276/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 35 out of 35 jobs passed in 51 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/2869f8eda2d325d0c4cd097a97276d3285a92481">2869f8eda2d325d0c4cd097a97276d3285a92481</a></p><details><summary><strong>ci:test:stable - 35 out of 35 jobs PASSED</strong></summary><details><summary><strong>35 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest(master)/10276/">test-portal-acceptance-pullrequest(master)</a></li><li>build-iw<ul><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1025835/">build-iw/0/0</a></li></ul></li><li>central-requirements<ul><li><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/477050/">central-requirements/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57-stable<ul><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/699452/">functional-smoke-tomcat90-mysql57-stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57-stable<ul><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1025837/">functional-upgrade-tomcat90-mysql57-stable/0/0</a></li><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1104026/">functional-upgrade-tomcat90-mysql57-stable/0/1</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1052546/">functional-upgrade-tomcat90-mysql57-stable/0/2</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1216666/">functional-upgrade-tomcat90-mysql57-stable/0/3</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1025836/">functional-upgrade-tomcat90-mysql57-stable/0/4</a></li></ul></li><li>js-unit<ul><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1122858/">js-unit/0/0</a></li><li><a href="https://test-1-29.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1005031/">js-unit/1/0</a></li></ul></li><li>modules-compile<ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1001140/">modules-compile/0/0</a></li><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1067421/">modules-compile/0/1</a></li><li><a href="https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/963095/">modules-compile/0/2</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1001139/">modules-compile/0/3</a></li><li><a href="https://test-1-33.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/941333/">modules-compile/0/4</a></li><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1067420/">modules-compile/0/5</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1115853/">modules-compile/0/6</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1115852/">modules-compile/0/7</a></li><li><a href="https://test-1-36.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1052545/">modules-compile/0/8</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/699454/">modules-compile/0/9</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/699453/">modules-compile/0/10</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1122914/">modules-compile/0/11</a></li><li><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/961193/">modules-compile/0/12</a></li></ul></li><li>modules-integration-mysql57-stable<ul><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/762991/">modules-integration-mysql57-stable/0/0</a></li><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1025838/">modules-integration-mysql57-stable/0/1</a></li><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1007266/">modules-integration-mysql57-stable/0/2</a></li><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1216667/">modules-integration-mysql57-stable/0/3</a></li></ul></li><li>modules-unit-stable<ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1216669/">modules-unit-stable/0/0</a></li></ul></li><li>playwright-compile<ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1216668/">playwright-compile/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57-stable<ul><li><a href="https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1025834/">playwright-js-tomcat90-mysql57-stable/0/0</a></li></ul></li><li>rest-builder<ul><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1231148/">rest-builder/0/0</a></li></ul></li><li>service-builder<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1007207/">service-builder/0/0</a></li></ul></li><li>unit-stable<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1007267/">unit-stable/0/0</a></li></ul></li><li>workspaces-compile<ul><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1115787/">workspaces-compile/0/0</a></li></ul></li></ul></details></details><h5>For more details click <a href="https://test-1-30.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/10276/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest(master)/10276/">test-portal-acceptance-pullrequest(master)#10276</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-30.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/10276/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>stable<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-content-management/liferay-portal/pull/5938">liferay-content-management#5938</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/43070613">[master] ci:test:stable - marcogalluzzi &gt; liferay-content-management - PR#5938 - 2024-10-16[02:20:45]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>43070613<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-30.liferay.com/job/publish-testray-report/24461/">publish-testray-report#24461</a><br/></div></div>

marcogalluzzi commented:
ci:forward:force

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` if the following test suites complete:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
AND If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
Skipping previously completed test suites:
- `ci:test:relevant`
- `ci:test:sf`
AND Skipping previously passed test suites:
- `ci:test:stable`

liferay-continuous-integration commented:
All required test suite(s) completed.
Forwarding pull request to `brianchandotcom`.
[Console](https://test-1-27.liferay.com/job/forward-pullrequest/1423//consoleFull)

liferay-continuous-integration commented:
Pull request has been successfully forwarded to  https://github.com/brianchandotcom/liferay-portal/pull/155219
[Console](https://test-1-27.liferay.com/job/forward-pullrequest/1423//consoleFull)
